File: 74-search-a-2d-matrix/74-search-a-2d-matrix.go
func searchMatrix(matrix [][]int, target int) bool {
 
    for i := 0; i < len(matrix); i++ {
        j := len(matrix[i]) - 1
        
        if matrix[i][j] == target {
            return true
        }
        
        if matrix[i][j] > target {
            for j = 0; j < len(matrix[i]); j++ {
                if matrix[i][j] == target {
                    return true
                } 
            }     
        }       
    }
    return false
}

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for arr in matrix:
            temp = arr
            if target > arr[-1]:
                continue
            elif target == arr[-1]:
                return True
            else:
                if self.binarySearch(temp, target):
                    return True
        return False
        
    def binarySearch(self, arr, item):
        i = 0
        j = len(arr) - 1
        
        while i <= j:
            mid = (i + j)//2
            
            if item > arr[mid]:
                i = mid + 1
            elif item < arr[mid]:
                j = mid - 1
            else:
                return True
            
        return False
        